modules/objdetect/src/qrcode.cpp

```diff
@@ -1245,5 +1228,1568 @@ cv::String QRCodeDetector::detectAndDecode(InputArray in,
+                                hull[extra_index_hull], hull[extra_next_index_hull]);
+            test_result_angle_list[2]
+            = intersectionLines(hull[extra_index_hull], hull[extra_next_index_hull],
+                                hull[index_hull], hull[next_index_hull]);
+            test_result_angle_list[3]
+            = intersectionLines(hull[index_hull], hull[next_index_hull],
+                                result_side_begin[0], result_side_end[0]);
+
+            const double test_diff_area
+                = fabs(fabs(contourArea(test_result_angle_list)) - experimental_area);
+            if (min_diff_area > test_diff_area)
+            {
+                min_diff_area = test_diff_area;
+                for (size_t i = 0; i < test_result_angle_list.size(); i++)
+                {
+                    result_angle_list[i] = test_result_angle_list[i];
+                }
+            }
+
+            extra_index_hull = extra_next_index_hull;
+        }
+        while(extra_index_hull != unstable_pnt);
+
+        index_hull = next_index_hull;
+    }
+    while(index_hull != unstable_pnt);
+
+    // check label points
+    if (norm(result_angle_list[0] - angle_list[1]) > 2) { result_angle_list[0] = angle_list[1]; }
+    if (norm(result_angle_list[1] - angle_list[0]) > 2) { result_angle_list[1] = angle_list[0]; }
+    if (norm(result_angle_list[3] - angle_list[2]) > 2) { result_angle_list[3] = angle_list[2]; }
+
+    // check calculation point
+    if (norm(result_angle_list[2] - angle_list[3]) >
+       (norm(result_angle_list[0] - result_angle_list[1]) +
+        norm(result_angle_list[0] - result_angle_list[3])) * 0.5 )
+    { result_angle_list[2] = angle_list[3]; }
+
+    return result_angle_list;
+}
+
+//      / | b
+//     /  |
+//    /   |
+//  a/    | c
+
+inline double MultipleQRDetect::getCosVectors(Point2f a, Point2f b, Point2f c)
+{
+    return ((a - b).x * (c - b).x + (a - b).y * (c - b).y) / (norm(a - b) * norm(c - b));
+}
+
+
+bool QRCodeDetector:: multipleDetect(InputArray in, OutputArrayOfArrays points) const
+{
+  Mat inarr = in.getMat();
+  CV_Assert(!inarr.empty());
+  CV_Assert(inarr.depth() == CV_8U);
+  if (inarr.cols <= 20 || inarr.rows <= 20)
+      return false;  // image data is not enough for providing reliable results
+
+  int incn = inarr.channels();
+  if( incn == 3 || incn == 4 )
+  {
+      Mat gray;
+      cvtColor(inarr, gray, COLOR_BGR2GRAY);
+      inarr = gray;
+  }
+  MultipleQRDetect qrdet;
+  qrdet.init(inarr, p->epsX, p->epsY);
+  if (!qrdet.localization()) { return false; }
+  vector< vector< Point2f > > pnts2f = qrdet.getTransformationPoints();
+  points.create((int)pnts2f.size(), 1, points.fixedType() ? points.type() : CV_32FC2);
+
+  for(int i = 0; i < (int)pnts2f.size(); i++)
+  {
+      points.create(4, 1, points.fixedType() ? points.type() : CV_32FC2, i, true);
+      Mat m = points.getMat(i);
+
+      switch(points.type())
+      {
+          case 10:
```

Comment:
10?

How do you plan to refactor code about OpenCV types?
Magic and hard-coded values must be avoided.

Consider using `.convertTo()`.